test/jump_tests.c

```diff
@@ -0,0 +1,61 @@
+#include <criterion/criterion.h>
+#include <criterion/new/assert.h>
+#include <criterion/parameterized.h>
+
+#include <stdbool.h>
+
+#include "fixtures/cpu_mmu.h"
+#include "common/util.h"
+
+typedef struct JrNZTestParams {
+    uint8_t opcode;
+    int8_t immediate_value;
+    Flag flag;
+    uint8_t flag_value;
+    bool branching_condition;
+} JrNZTestParams;
+
+ParameterizedTestParameters(jr_cc_n, jr_cc_n) {
+    static JrNZTestParams params[] = {
+  // take the branch
+        {0x20, 101, Z_FLAG, 0, true },
+        {0x30, 101, C_FLAG, 0, true },
+        {0x28, 101, Z_FLAG, 1, true },
+        {0x38, 101, C_FLAG, 1, true },
+
+ // don't take the branch
+        {0x20, 101, Z_FLAG, 1, false},
+        {0x30, 101, C_FLAG, 1, false},
+        {0x28, 101, Z_FLAG, 0, false},
+        {0x38, 101, C_FLAG, 0, false},
```

Comment:
Maybe we should do some tests with negative offsets? 🤔

Reply:
Do you mean the immediate value?

Reply:
Exactly